app/forms/csv_upload_form.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CSVUploadForm
+  include ActiveModel::Validations
+
+  attr_reader :file
+
+  validates_with CSVValidator
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(file)
+    @file = file
+    return false unless file && valid?
+
+    begin
+      save_keywords
+    rescue ActiveRecord::StatementInvalid
+      errors.add(:base, I18n.t('csv.validation.bad_keyword_length'))
+    end
+    errors.empty?
+  end
+
+  private
+
+  def save_keywords
+    keyword_hash.each do |hash|
+      keyword = Keyword.find_or_create_by!(hash)
+      keyword_user_hash = { keyword: keyword, user: @user }
+      KeywordUser.find_or_create_by!(keyword_user_hash)
+    end
+  end
+
+  def keyword_hash
+    CSV.read(file).filter_map { |row|
+      name = row.join(',')
+      name.blank? ? false : { name: name }
+    }
```

Comment:
How do we store the keywords? 👀 

From my reading, I see that a `keyword` object will have the column `name` with a value like `keywoard1,keyword2,keyword3,keyword4,keyword5`. Is that correct? 💭 

If yes, I recommend 2 options:
- either keep the same format, but we must rename `keyword` with `keyword_batch` or `keyword_set` or `keyword_group`
- either see if we can change the structure so that 1 keyword represent only 1 keyword (instead of a list of keywords).

This is mainly to ensure the name of our variables correctly describe what they hold.

Reply:
The keyword is saved individually, the join logic here is to connects keyword in the format `hello, world` > `['hello', 'world'] > `'hello, world'`.
The idea of the DB is `1 keyword represent only 1 keyword`.

Reply:
My bad, at the time I wrote this I though csv would be a single line of multiple keywords (instead of 1 keyword per line) 😇